data-prepper-core/src/main/java/com/amazon/dataprepper/plugin/PluginArgumentsContext.java

```diff
@@ -64,11 +66,22 @@ private PluginArgumentsContext(final Builder builder) {
         if(typedArgumentsSuppliers.containsKey(parameterType)) {
             return typedArgumentsSuppliers.get(parameterType);
         }
-        else if (beanFactory != null) {
-            return () -> beanFactory.getBean(parameterType);
+        else {
+            return createBeanSupplier(parameterType);
+        }
+    }
+
+    private Supplier<Object> createBeanSupplier(final Class<?> parameterType) {
```

Comment:
I think it would be better if all of this behavior happened in the `Supplier`. Otherwise the call to `getBean` happens before it is needed.

Reply:
@dlvenable I originally implemented as you suggested. I found when the supplier is actually called, if the bean is unavailable, the error message is very unreadable. Data Prepper log an exception saying no valid pipeline found. Attempting to get the bean early shows the exception in context of the plugin that cannot not be created.

I can try moving the getBean call back to inside the supplier and make some changes to how exceptions are handled when building the pipeline.

Reply:
@dlvenable I tested moving the try block to within the supplier and everything worked as intended. I'll push an update soon after I add some test cases for the new functionallity